lucene/core/src/java/org/apache/lucene/index/PointValues.java

```diff
@@ -398,8 +405,8 @@ private long estimatePointCount(IntersectVisitor visitor, PointTree pointTree)
         if (pointTree.moveToChild()) {
           long cost = 0;
           do {
-            cost += estimatePointCount(visitor, pointTree);
-          } while (pointTree.moveToSibling());
+            cost += estimatePointCount(visitor, pointTree, upperBound - cost);
+          } while (cost <= upperBound && pointTree.moveToSibling());
```

Comment:
I believe that we can stop counting if `cost == upperBound`?

```suggestion
          } while (cost < upperBound && pointTree.moveToSibling());
```